pkg/reconciler/stats_reporter_test.go

```diff
@@ -64,32 +69,37 @@ func TestReporter_ReportDuration(t *testing.T) {
 
 	latency := getMetric(t, ServiceReadyLatencyN)
 	if v := latency.Data.(*view.LastValueData).Value; v != 1000 {
-		t.Errorf("expected latency %v, Got %v", 1000, v)
+		t.Errorf("Expected latency %v, Got %v", 1000, v)
 	}
 	checkTags(t, expectedTags, latency.Tags)
 
 	count := getMetric(t, ServiceReadyCountN)
-	if v := count.Data.(*view.CountData).Value; v != 1 {
-		t.Errorf("expected latency %v, Got %v", 1, v)
+	if got, want := count.Data.(*view.CountData).Value, countWas+1; got != want {
+		t.Errorf("Latency report count = %d, want: %d", got, want)
 	}
 	checkTags(t, expectedTags, count.Tags)
 }
 
 func getMetric(t *testing.T, metric string) *view.Row {
+	t.Helper()
 	rows, err := view.RetrieveData(metric)
 	if err != nil {
-		t.Errorf("failed retrieving data: %v", err)
+		t.Errorf("Failed retrieving data: %v", err)
+	}
+	if len(rows) == 0 {
+		return nil
```

Comment:
Do we need to add nil checks to all `getMetrics` calls, like the one you added?

Reply:
Not really, this is only required for the initial call, when nothing has yet been reported. The way test is structured, rest of it will have the data.